# !/usr/bin/bash
#------------------------------------------------------------------------------
# SNPs filter Analysis        
# Data after alignment, mark duplication and GATK remapping
#
# Link to publication
# TO ADD ONCE AVAILABLE
#
# Script available from:
# https://github.com/CTR-BFX/Blake_Watson
#
#
# Analysis Performed by Xiaohui Zhao
# CTR Bioinformatics Facility
# Centre for Trophoblast Reseach, University of Cambridge, UK
# Copyright Xiaohui Zhao (xz289@cam.ac.uk)
#
#------------------------------------------------------------------------------
# License Information
# This program is free software: you can redistribute it and/or modify  
# it under the terms of the GNU General Public License as published by  
# the Free Software Foundation, version 3.
#
# This program is distributed in the hope that it will be useful, but 
# WITHOUT ANY WARRANTY; without even the implied warranty of 
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU 
# General Public License for more details.
#
# You should have received a copy of the GNU General Public License 
# along with this program. If not, see <http://www.gnu.org/licenses/>.
#------------------------------------------------------------------------------

module load GenomeAnalysisTK/3.7
module load java/1.8.0_111
module load bcftools/1.3.1
module load vcftools/0.1.15
module load snpEff/4.3t
module load R/3.4.2

samplepath=/storage/CTR-Projects/CTR_edw23/CTR_edw23_0001
samplenames='recalibrated'
samplepath=/storage/CTR-Projects/CTR_edw23/CTR_edw23_0001/recalibration
sampleoutpath=/home/xz289/CTR_edw23_0001/Recalibration


#---------------------------------------------------------------------------------#
#       Merge 8 samples first and then do the filter                              #
#---------------------------------------------------------------------------------#

samplenames='recalibrated'
samplepath=/storage/CTR-Projects/CTR_edw23/CTR_edw23_0001/recalibration
sampleoutpath=/home/xz289/CTR_edw23_0001/Recalibration
for name in $samplenames;
do

  egrep -v "^#" ${samplepath}/${name}_variants.vcf \
    | wc -l
  
  bgzip -c ${samplepath}/${name}_variants.vcf \
    > ${sampleoutpath}/${name}.vcf.gz
  
  tabix -p vcf ${sampleoutpath}/${name}.vcf.gz
  
  bcftools stats -F ${refpath}/Mus_musculus.GRCm38.dna.chromosome.all.fa \
    -s- ${sampleoutpath}/${name}.vcf.gz > ${sampleoutpath}/${name}.vcf.gz.stats
  
  mkdir ${sampleoutpath}/${name}_plots
  
  plot-vcfstats -p ${sampleoutpath}/${name}_plots/ ${sampleoutpath}/${name}.vcf.gz.stats

#----------------------------------------------------------------------------------#
#     filter out the snps based on Oey (2015) paper for merged vcf file            #
#                            Quality Filters                                       #                              
#----------------------------------------------------------------------------------#

  zcat ${sampleoutpath}/${name}.vcf.gz | vcf-annotate -H -f \
    +/1=0.0001/2=0.002/3=0.00001/4=0.0001/a=6/-c/-r/d=14/D=100/q=25/Q=40/W=3/w=5/H=0.0001/v=0 \
    | bgzip -c > ${sampleoutpath}/${name}_vcf_filter1.vcf.gz

## middle step to check the vcf file SNVs.
  bcftools stats -s- ${sampleoutpath}/${name}_vcf_filter1.vcf.gz > \
    ${sampleoutpath}/${name}_vcf_filter1.stats.txt
  
  mkdir ${sampleoutpath}/${name}_vcf_filter1_plots
  
  plot-vcfstats  -p ${sampleoutpath}/${name}_vcf_filter1_plots/ \
    ${sampleoutpath}/${name}_vcf_filter1.stats.txt
done;



cd $sampleoutpath;

gunzip recalibrated_vcf_filter1.vcf.gz


## The recalibrated_exclude_locs.txt was generated by Rscript.

Rscript /Users/xz289/Documents/CTR_edw23_0001/Script/Homo_Dinu_SegDup_Het.R

vcftools --vcf recalibrated_vcf_filter1.vcf \
         --exclude-positions recalibrated_exclude_locs.txt \
         --recode \
         --recode-INFO-all \
         --out recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14

bgzip -c recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14.recode.vcf \
  > recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14.recode.vcf.gz

tabix -p vcf recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14.recode.vcf.gz

bcftools stats -F ${refpath}/Mus_musculus.GRCm38.dna.chromosome.all.fa \
  -s-  recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14.recode.vcf.gz \
  > recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14.recode.vcf.gz.stats

mkdir recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14_plots

plot-vcfstats -p recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14_plots/ \
  recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14.recode.vcf.gz.stats

#---------------------------------------------------------------------------------#
# snpEff annotation                                                               #
#---------------------------------------------------------------------------------#

java -Xmx4g -jar ${snpEffdir}/snpEff.jar \
     -c ${snpEffdir}/snpEff.config \
     -v GRCm38.86 recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14.recode.vcf \
      > recalibrated_vcf_filter1_segDup_simpRL9_hete3_RefH8D14.ann.vcf
      
#----------------------         FIN  --------------------------------------------#
